src/service-catalog-appregistry-stack.ts

```diff
@@ -0,0 +1,83 @@
+/**
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+import { Application } from '@aws-cdk/aws-servicecatalogappregistry-alpha';
+import { Aws, CfnCondition, CfnOutput, CfnParameter, CfnResource, Fn, Stack, Tags } from 'aws-cdk-lib';
+import { StackProps } from 'aws-cdk-lib/core/lib/stack';
+import { Construct } from 'constructs';
+import { OUTPUT_SERVICE_CATALOG_APPREGISTRY_APPLICATION_ARN } from './common/constant';
+import { SolutionInfo } from './common/solution-info';
+
+export interface ServiceCatalogAppRegistryProps extends StackProps {
+}
+
+export class ServiceCatalogAppregistryStack extends Stack {
+  constructor(scope: Construct, id: string, props?: ServiceCatalogAppRegistryProps) {
+    super(scope, id, props);
+
+    const featureName = 'AppRegistry';
+    this.templateOptions.description = `(${SolutionInfo.SOLUTION_ID}-appreg ${SolutionInfo.SOLUTION_NAME} - ${featureName} ${SolutionInfo.SOLUTION_VERSION_DETAIL}`;
+
+    const projectIdParam = new CfnParameter(this, 'ProjectId', {
```

Comment:
use the common method to create projectId parameter with proper validation rule

Reply:
done

Reply:
use [this one](https://github.com/awslabs/clickstream-analytics-on-aws/blob/cb0755b5e79f57a7351ca92b548ff0a5a399064d/src/common/parameters.ts#L429) instead